src/main/java/me/moodcat/api/RoomAPI.java

```diff
@@ -174,4 +174,28 @@ public NowPlaying getCurrentTime(@PathParam("id") final int roomId) {
         return nowPlaying;
     }
 
+    /**
+     * Process a vote to a song. A vote is either "like" or "dislike".
+     *
+     * @param vote
+     *            The vote.
+     * @return The song object, if the process was succesful.
+     * @throws InvalidVoteException
+     *             Thrown if an invalid vote was cast.
+     */
+    @POST
+    @Path("{id}/vote/{vote}")
+    @Transactional
+    public RoomModel voteSong(@PathParam("id") final int roomId,
+            @PathParam("vote") final String vote) throws InvalidVoteException {
+        final RoomInstance roomInstance = this.backend.getRoomInstance(roomId);
+
+        if ("like".equals(vote) || "dislike".equals(vote)) {
```

Comment:
See:
> @PathParam, @QueryParam, @MatrixParam, @FormParam, and @HeaderParam are represented as strings in a raw HTTP request. *The specification says that these types of injected parameters can be converted to objects if these objects have a valueOf(String) static method or a constructor that takes one String parameter*. 
from :  https://docs.jboss.org/resteasy/docs/2.2.1.GA/userguide/html/StringConverter.html

And:
> valueOf
>
> ```java
> public static <T extends Enum<T>> T valueOf(Class<T> enumType,                                            String name)
> ```
> Returns the enum constant of the specified enum type with the specified name. The name must match exactly an identifier used to declare an enum constant in this type. (Extraneous whitespace characters are not permitted.)
> Throws:
> * `IllegalArgumentException` - if the specified enum type has no constant with the specified name, or the specified class object does not represent an enum type
? * `NullPointerException` - if enumType or name is null
from: https://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Enum.html

Sounds the most consistent to me.